libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs

```diff
@@ -255,8 +255,14 @@ private static bool IsTrustedUrl(Uri uri, ILogger logger)
             }
         }
 
-        private static bool ShouldSetToken(HttpRequestMessage request, ILogger logger)
+        private bool ShouldSetToken(HttpRequestMessage request, ILogger logger)
         {
+            if (string.IsNullOrEmpty(MicrosoftAppId) || MicrosoftAppId == AuthenticationConstants.AnonymousSkillAppId)
+            {
+                // We don't set the token if the AppId is not set.
```

Comment:
Optional: maybe a bit more info for future devs

```suggestion
                // We don't set the token if the AppId is not set, since it means that we are in an un-authenticated scenario
```

Reply:
@johnataylor and I still want to remove the TrustServiceUrl concept; by the time the bot gets to `ShouldSetToken()`, it's already processed the request and has decided to send an activity.

@johnataylor's proposal to remove the TrustServiceUrl concept was to have `ShouldSetToken()` always return true.

Implementing this proposal verbatim likely won't hold as `ProcessHttpRequestAsync()` will likely try to fetch a token with null values, leading to a NullReferenceException or something else. But the proposal is elegant and what I believe we should move forward with.